src/Console/Program.cs

```diff
@@ -16,10 +17,18 @@ static async Task<int> Main(string[] args)
             rootCommand.AddCommand(new DeployCommand());
             rootCommand.AddCommand(new LoginCommand());
             rootCommand.AddCommand(new PublishCommand());
-            rootCommand.AddCommand(new PublishPackageCommand());
-           
+            rootCommand.AddCommand(new PublishPackageCommand());
+
+            // cannot have an argument associated with --name with spaces. This condition is required to continue allowing users to have --name as a parameter
+            if (Array.IndexOf(args, "--name") > -1)
+            {
+                args[Array.IndexOf(args, "--name")] = "--id";
+            }

```

Comment:
Instead of this change (although it works, it's not pretty) I propose to change the way we invoke the Handlers.
Since we have several invocations with a lot of parameters, we need to use custom types. But the idea is as follows (example for DeployCommand):
```csharp
    ...
    Handler = CommandHandler.Create((DeployParameters x) => DeployHandler(x));
}

public class DeployParameters
{
    public bool Verbose { get; set; }
    public string CustomerInfrastructureName { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public FileInfo Parameters { get; set; }
    public string Type { get; set; }
    public string Site { get; set; }
    public string License { get; set; }
    public string Package { get; set; }
    public string Target { get; set; }
    public string TemplateName { get; set; }
    public DirectoryInfo Output { get; set; }
    public string[] ReplaceTokens { get; set; }
    public bool Interactive { get; set; }
    public bool TerminateOtherVersions { get; set; }
}

public async Task DeployHandler(DeployParameters parameters)
{
    System.Console.WriteLine(parameters.Name);
    // get new environment handler and run it
    CreateSession(parameters.Verbose);
    NewEnvironmentHandler newEnvironmentHandler = ServiceLocator.Get<NewEnvironmentHandler>();
    await newEnvironmentHandler.Run(
        parameters.Name,
        parameters.Parameters,
        (EnvironmentType)Enum.Parse(typeof(EnvironmentType), parameters.Type),
        parameters.Site,
        parameters.License,
        parameters.Package,
        (DeploymentTarget)Enum.Parse(typeof(DeploymentTarget), parameters.Target),
        parameters.Output,
        parameters.ReplaceTokens,
        parameters.Interactive,
        parameters.CustomerInfrastructureName,
        parameters.Description,
        parameters.TemplateName,
        parameters.TerminateOtherVersions,
        false
    );
}
```